File: android/app/src/main/kotlin/ai/gowtam/oak/features/usage/UsageFormat.kt
package ai.gowtam.oak.features.usage

import java.text.DateFormat
import java.util.Date
import java.util.Locale
import java.util.TimeZone

/** Medium date + short time in [timeZone] (device zone by default). Never a trailing `Z`. */
internal fun formatUsageFetchedAt(
    ms: Long,
    timeZone: TimeZone = TimeZone.getDefault(),
    locale: Locale = Locale.getDefault(),
): String {
    val fmt = DateFormat.getDateTimeInstance(DateFormat.MEDIUM, DateFormat.SHORT, locale)
    fmt.timeZone = timeZone
    return fmt.format(Date(ms))
}

internal data class UsageAttributionParts(val source: String, val legal: String?)

/** Split `"championsbattledata.com — a community-maintained …"` into name + remainder. */
internal fun parseUsageAttribution(raw: String): UsageAttributionParts {
    val trimmed = raw.trim()
    if (trimmed.isEmpty()) return UsageAttributionParts(trimmed, null)
    for (sep in listOf(" — ", " – ", " - ")) {
        val idx = trimmed.indexOf(sep)
        if (idx <= 0) continue
        val source = trimmed.substring(0, idx).trim()
        val legal = trimmed.substring(idx + sep.length).trim()
        if (source.isNotEmpty()) {
            return UsageAttributionParts(source, legal.takeIf { it.isNotEmpty() })
        }
    }
    return UsageAttributionParts(trimmed, null)
}

File: ios/OakApp/Features/Usage/UsageFormat.swift
import Foundation

/// Medium date + short time in the given zone (device zone by default).
/// Never ISO-8601 / trailing `Z` — CF-USAGE-AC-1.3 as-of stamp, human-readable.
func formatUsageFetchedAt(
  _ ms: Int64,
  timeZone: TimeZone = .current,
  locale: Locale = .current
) -> String {
  let date = Date(timeIntervalSince1970: TimeInterval(ms) / 1000)
  let formatter = DateFormatter()
  formatter.dateStyle = .medium
  formatter.timeStyle = .short
  formatter.timeZone = timeZone
  formatter.locale = locale
  formatter.doesRelativeDateFormatting = false
  return formatter.string(from: date)
}

struct UsageAttributionParts: Equatable, Sendable {
  var source: String
  var legal: String?
}

/// Split the T15 attribution blob into a short source name and optional legal line.
/// `"championsbattledata.com — a community-maintained …"` → name + remainder.
func parseUsageAttribution(_ raw: String) -> UsageAttributionParts {
  let trimmed = raw.trimmingCharacters(in: .whitespacesAndNewlines)
  guard !trimmed.isEmpty else { return UsageAttributionParts(source: trimmed, legal: nil) }
  for sep in [" — ", " – ", " - "] {
    if let range = trimmed.range(of: sep) {
      let source = String(trimmed[..<range.lowerBound])
        .trimmingCharacters(in: .whitespacesAndNewlines)
      let legal = String(trimmed[range.upperBound...])
        .trimmingCharacters(in: .whitespacesAndNewlines)
      if !source.isEmpty {
        return UsageAttributionParts(source: source, legal: legal.isEmpty ? nil : legal)
      }
    }
  }
  return UsageAttributionParts(source: trimmed, legal: nil)
}
